app/src/behaviors/behavior_combo.c

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_combo
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+#include <sys/dlist.h>
+#include <kernel.h>
+
+#include <zmk/behavior.h>
+#include <zmk/matrix.h>
+#include <zmk/endpoints.h>
+#include <zmk/event-manager.h>
+#include <zmk/events/position-state-changed.h>
+#include <zmk/events/keycode-state-changed.h>
+#include <zmk/events/modifiers-state-changed.h>
+#include <zmk/hid.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+struct combo_t {
```

Comment:
Only should have `_t` if typedef'd

Reply:
I'm often using 'combo' as the name of a variable, so it's kind of nice to have it like that (and not shadowing the type with a variable name). Thoughts on good naming? I could try to find other names for the variables too, like 'candidate' instead of 'combo'.